htsinfer/get_library_type.py

```diff
@@ -159,57 +158,66 @@ def _align_mates(self):
         previous_seq_id2 = None
 
         reads1 = []  # List to store alignments for one read from file1
-        mate1 = []  # List to store alignments for each read
+        mate1 = []  # List to store alignments for each read from file1
         reads2 = []  # List to store alignments for one read from file2
+        mate2 = []  # List to store alignments for each read from file2
 
         concordant = 0
 
         for read1 in samfile1:
-            if not read1.flag & (1 << 2) and isinstance(read1.query_name, str):
-                seq_id1 = read1.query_name
-                if (
-                    seq_id1 != previous_seq_id1 and
-                    previous_seq_id1 is not None
-                ):
-                    mate1.append(reads1.copy())
-                    reads1.clear()
-                if read1.reference_end:
-                    reads1.append(read1)
+            seq_id1 = read1.query_name
+            if (
+                seq_id1 != previous_seq_id1 and
+                previous_seq_id1 is not None
+            ):
+                mate1.append(reads1.copy())
```

Comment:
The solution still requires a lot of memory. It's fine but please open a low priority issue to improve this implementation at some point in the future.

Reply:
Opened #158